src/curve/affine.rs

```diff
@@ -0,0 +1,686 @@
+    #[inline]
+    pub fn multiply(&self, by: &[u8; 32]) -> AffinePoint {
+        JacobianPoint::multiply(&self.into(), by).into()
+    }
+
+    /// Performs an affine scalar multiplication from `by`
+    /// given as byte representation of a `Scalar` element.
+    ///
+    /// **This operation is variable time with respect
+    /// to the scalar.** If the scalar is fixed,
+    /// this operation is effectively constant time.
+    #[inline]
+    pub fn multiply_vartime(&self, by: &[u8; 32]) -> AffinePoint {
+        JacobianPoint::multiply_vartime(&self.into(), by).into()
+    }
+
+    /// Performs the affine sum [`by_lhs` * `self` + `by_rhs` * `rhs`] with `by_lhs`
+    /// and `by_rhs` given as byte representations of `Scalar` elements.
+    #[inline]
+    pub fn multiply_double(
+        &self,
+        rhs: &AffinePoint,
+        by_lhs: &[u8; 32],
+        by_rhs: &[u8; 32],
+    ) -> AffinePoint {
+        JacobianPoint::multiply_double(&self.into(), &rhs.into(), by_lhs, by_rhs).into()
+    }
+
+    /// Performs the affine sum [`by_lhs` * `self` + `by_rhs` * `rhs`] with `by_lhs`
+    /// and `by_rhs` given as byte representations of `Scalar` elements.
+    ///
+    /// **This operation is variable time with respect
+    /// to the scalars.** If the scalars are fixed,
+    /// this operation is effectively constant time.
+    #[inline]
+    pub fn multiply_double_vartime(
+        &self,
+        rhs: &AffinePoint,
+        by_lhs: &[u8; 32],
+        by_rhs: &[u8; 32],
+    ) -> AffinePoint {
+        JacobianPoint::multiply_double_vartime(&self.into(), &rhs.into(), by_lhs, by_rhs).into()
+    }
+
+    /// Performs the affine sum [`by_self` * `self` + `by_basepoint` * `ODD_MULTIPLES_BASEPOINT`]
+    /// with `by_self` and `by_rby_basepoinths` given as byte representations of `Scalar` elements.
+    ///
+    /// **This operation is variable time with respect
+    /// to the scalars.** If the scalars are fixed,
+    /// this operation is effectively constant time.
+    #[inline]
+    pub fn multiply_double_with_basepoint_vartime(
+        &self,
+        by_self: &[u8; 32],
+        by_basepoint: &[u8; 32],
+    ) -> AffinePoint {
+        JacobianPoint::multiply_double_with_basepoint_vartime(&self.into(), by_self, by_basepoint)
+            .into()
+    }
+
+    /// Multiplies by the curve cofactor
+    pub fn clear_cofactor(&self) -> AffinePoint {
+        let point: JacobianPoint = self.into();
+
+        point.clear_cofactor().into()
+    }
+
+    /// Returns true if this point is free of an $h$-torsion component.
+    /// This should always return true unless an "unchecked" API was used.
+    pub fn is_torsion_free(&self) -> Choice {
+        let point: JacobianPoint = self.into();
+
+        point.is_torsion_free()
+    }
+}
+
+impl<'a, 'b> Mul<&'b Scalar> for &'a AffinePoint {
+    type Output = JacobianPoint;
+
+    fn mul(self, other: &'b Scalar) -> Self::Output {
+        JacobianPoint::from(self).multiply(&other.to_bytes())
```

Comment:
Is there a reason to prefer this form over `JacobianPoint::multiply(&self.into(), &other.to_bytes())`?

Reply:
No, just noticed it being different from e.g. `JacobianPoint::multiply_double_with_basepoint_vartime(&self.into(), by_self, by_basepoint)` on line 400 above and wondered if there was an advantage I was missing.